app/ts/utils/utilities.ts

```diff
@@ -194,10 +196,97 @@ export const getStakeInOutcome = async (reader: AccountAddress, market: AccountA
 	})
 }
 
-export const getStakesOnAllOutcomesOnYesNoMarketOrCategorical = async (reader: AccountAddress, market: AccountAddress, numOutcomes: number, numTicks: EthereumQuantity) => {
-	const allPayoutNumeratorCombinations = Array.from({ length: numOutcomes }, (_, outcome) =>
+export const getAllPayoutNumeratorCombinations = (numOutcomes: number, numTicks: EthereumQuantity): readonly bigint[][] => {
+	return Array.from({ length: numOutcomes }, (_, outcome) =>
 		Array.from({ length: numOutcomes }, (_, index) => index === outcome ? numTicks : 0n)
 	)
+}
+
+export const getStakesOnAllOutcomesOnYesNoMarketOrCategorical = async (reader: AccountAddress, market: AccountAddress, numOutcomes: number, numTicks: EthereumQuantity) => {
+	const allPayoutNumeratorCombinations = getAllPayoutNumeratorCombinations(numOutcomes, numTicks)
 	const payoutDistributionHashes = await Promise.all(allPayoutNumeratorCombinations.map(async (payoutNumerator) => EthereumQuantity.parse(await derivePayoutDistributionHash(reader, market, payoutNumerator))))
 	return await Promise.all(payoutDistributionHashes.map((payoutDistributionHash) => getStakeInOutcome(reader, market, payoutDistributionHash)))
 }
+
+export const contributeToMarketDispute = async (writer: AccountAddress, market: AccountAddress, payoutNumerators: EthereumQuantity[], amount: EthereumQuantity, reason: string) => {
+	const client = createWriteClient(writer)
+	return await client.writeContract({
+		abi: MARKET_ABI,
```

Comment:
Is there a way to create something like a global `marketClient` that has the ABI already provided and parsed so we don't have to do it every time?  Presumably it would also have the address baked in, and it would give us type checked functions like `contribute(payoutNumerators, amount, reason)`.  I thought viem could do this?